tests/nested/image-build/task.yaml

```diff
@@ -2,58 +2,73 @@ summary: create ubuntu-core image and execute the suite in a nested qemu instanc
-
 prepare: |
-    . $TESTSLIB/nested.sh
-
     # determine arch related vars
-    set_vars
+    case "$NESTED_ARCH" in
+    amd64)
+        QEMU="$(which qemu-system-x86_64)"
+        ;;
+    i386)
+        QEMU="$(which qemu-system-i386)"
+        ;;
+    *)
+        echo "unsupported architecture"
+        exit 1
+        ;;
+    esac
 
     # create ubuntu-core image
-    mkdir -p $WORK_DIR
-    snap install --devmode --edge ubuntu-image
-    /snap/bin/ubuntu-image --image-size 3G $TESTSLIB/assertions/$model_file --channel $CORE_CHANNEL --output ubuntu-core.img
-    mv ubuntu-core.img $WORK_DIR
+    mkdir -p /tmp/work-dir
+    snap install --devmode --beta ubuntu-image
+    /snap/bin/ubuntu-image --image-size 3G $TESTSLIB/assertions/nested-${NESTED_ARCH}.model --channel $CORE_CHANNEL --output ubuntu-core.img
+    mv ubuntu-core.img /tmp/work-dir
 
     # create assertion block device
     genisoimage -volid cidata -joliet -rock -o assertions.disk $TESTSLIB/assertions/auto-import.assert
 
-    #setup_bridge $TAP_DEVICE
-
     . $TESTSLIB/systemd.sh
-    systemd_create_and_start_unit nested-vm "$vm_unit_command"
+    systemd_create_and_start_unit nested-vm "${QEMU} -m 1024 -nographic -net nic,model=virtio -net user,hostfwd=tcp::8022-:22 -drive file=/tmp/work-dir/ubuntu-core.img,if=virtio,cache=none -drive file=${PWD}/assertions.disk,if=virtio,cache=none"
 
 restore: |
     snap remove ubuntu-image
     . $TESTSLIB/systemd.sh
     systemd_stop_and_destroy_unit nested-vm
-    rm -rf $WORK_DIR
-
-    #. $TESTSLIB/nested.sh
-    #teardown_bridge $TAP_DEVICE
+    rm -rf /tmp/work-dir
 
 execute: |
-    . $TESTSLIB/nested.sh
+    execute_remote(){
+        sshpass -p ubuntu ssh -p 8022 -q -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no user1@localhost "$*"
+    }
+
+    prepare_ssh(){
+        execute_remote "sudo adduser --extrausers --quiet --disabled-password --gecos '' test"
+        execute_remote "echo test:ubuntu | sudo chpasswd"
+        execute_remote "echo 'test ALL=(ALL) NOPASSWD:ALL' | sudo tee /etc/sudoers.d/test-user"
+    }
+
+    wait_for_ssh(){
+        retry=300
+        while ! execute_remote true; do
+            retry=$(( retry - 1 ))
+            if [ $retry -le 0 ]; then
+                echo "Timed out waiting for ssh. Aborting!"
+                exit 1
+            fi
+            sleep 1
+        done
+    }
 
     wait_for_ssh
 
     prepare_ssh
 
     cd $SPREAD_PATH
 
-    curl -s -O https://niemeyer.s3.amazonaws.com/spread-amd64.tar.gz && tar xzvf spread-amd64.tar.gz && rm -f spread-amd64.tar.gz && mv spread $WORK_DIR
+    curl -s -O https://niemeyer.s3.amazonaws.com/spread-amd64.tar.gz && tar xzvf spread-amd64.tar.gz && rm -f spread-amd64.tar.gz && mv spread /tmp/work-dir
 
-    $WORK_DIR/spread -v external:ubuntu-core-16-64:tests/main/ubuntu-core-reboot \
-                        external:ubuntu-core-16-64:tests/main/install-sideload \
```

Comment:
These are arbitrarily chosen tests. We need to reduce the overall run time of the test so it fits in the 50 minutes Travis gives us.